### PR TITLE
Set NodeRef to Machines

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
This PR sets the NodeRef on Machine's Spec effectively linking each created Machine to a Kubernetes node.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to #214 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/cloud/aws/actuators/machine/actuator.go [hunk 1]
```diff
@@ -410,5 +410,52 @@ func (a *Actuator) Exists(ctx context.Context, cluster *clusterv1.Cluster, machi
...
+
+	listOpt := metav1.ListOptions{}
+	for {
+		nodeList, err := coreClient.Nodes().List(listOpt)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to query cluster nodes")
+		}
+
+		for _, node := range nodeList.Items {
+			// TODO(vincepri): Improve this comparison without relying on substrings.
```
Comment (randomvariable) [inline pkg/cloud/aws/actuators/machine/actuator.go]: Create an issue for this straight after merge.

**Comment by randomvariable** [comment]
lgtm from me, seems in line with what was discussed in https://kubernetes.slack.com/archives/CD6U2V71N/p1550092770209500

**Comment by randomvariable** [comment]
Needs rebasing though

File: pkg/apis/awsprovider/v1alpha1/types.go [hunk 2]
```diff
@@ -189,6 +189,17 @@ func (s Subnets) ToMap() map[string]*SubnetSpec {
 	return res
 }
 
+// FilterID returns a single subnet matching the given id or nil.
+func (s Subnets) FilterID(id string) *SubnetSpec {
```
Comment (chuckha) [inline pkg/apis/awsprovider/v1alpha1/types.go]: filters return lists, I would expect this to be called FindByID or similar

File: pkg/cloud/aws/actuators/machine/actuator.go [hunk 3]
```diff
@@ -135,17 +135,24 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
...
 
 	var bootstrapToken string
 	if isNodeJoin {
-		bootstrapToken, err = a.getNodeJoinToken(cluster)
+		coreClient, err := a.coreV1Client(cluster)
+		if err != nil {
+			return errors.Wrapf(err, "failed to retrieve corev1 client for cluster %q", cluster.Name)
+		}
+
+		bootstrapToken, err = tokens.NewBootstrap(coreClient, 10*time.Minute)
```
Comment (chuckha) [inline pkg/cloud/aws/actuators/machine/actuator.go]: the 10 should be a constant, `bootstrapTokenTTL` maybe?

**Comment by chuckha** [comment]
hello I am 🐌

**Pull request merged**